app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt

```diff
@@ -19,13 +19,15 @@ import android.widget.TextView
 import android.view.inputmethod.InputMethodManager
 import android.net.ConnectivityManager
 import kotlinx.android.synthetic.main.content_no_internet.view.*
+import org.fossasia.openevent.general.data.Preference
 import org.fossasia.openevent.general.utils.Utils
 
 
 class EventsFragment : Fragment() {
     private val eventsRecyclerAdapter: EventsRecyclerAdapter = EventsRecyclerAdapter()
     private val eventsViewModel by viewModel<EventsViewModel>()
     private lateinit var rootView: View
+    private val preference: Preference = Preference()
```

Comment:
No logic in fragment

Reply:
@iamareebjamal I moved the logic viewmodel in this commit https://github.com/fossasia/open-event-android/pull/172/commits/0636e281fd003e7976388320f3091bc858a825a8
but the tests were failing so I reverted.
Was it the correct way?

Reply:
Yes